CMSgov/mint-app
# [MINT-3020] Collaboration area page

# MINT-3020

## Description


- Created page and route `/collaboration-area`
- Created/updated `Breadcrumbs` component to have default/preset breadcrumbs
  - Implemented this updated component across all of MINT
  - Includes all breadcrumb updates for Collaboration Area
- Added better support for underlining the current tab based on the parent route
  - `getActiveTab` util to identify tab that should be underlined, addresses custom/edge cases
- Added `ShareExportButton` component to render a select dropdown to share, export or remove model plan
  - Share or export will preselect the chosen tab when opened
- Left some commented out code in preparation for future tickets - ex: document, crtdl, teams etc.  This was ported over from the task list.  
- Updated emails to correctly include collaboration base routes
- Updated unit tests
- Updated cypress tests


## How to test this change

1.  Log in as both/either admin on base user
2. Click on any model plan from the home page tables (All models, My models, etc)
3. Verify that it navigates to the collaboration area component (if have edit access)
4. Verify share export model works as expected
5. Verify all breadcrumbs match figma

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
